docs/core/middleware.md

```diff
@@ -0,0 +1,155 @@
+<!--
+order: 3
+-->
+
+# Middlewares
+
+
+`Middlewares` are objects that the developer can create to add logic before or after the transaction handler in a composable manner.
+{synopsis}
+
+## Pre-requisite Readings
+
+* [Anatomy of a Cosmos SDK Application](../basics/app-anatomy.md) {prereq}
+* [Transactons](transactions.md){prereq}
+
+
+## Middlewares
+
+The SDK Baseapp's implementation of ABCI CheckTx, DeliverTx, and Simulate methods use a middleware-based design. Middlewares can add logic to be executed before or after a transaction handler execution. Middlewares allow us to solve use cases like transaction Tips and refund unused gas. 
+
+
```

Comment:
In general, cosmos developers are used to the word "antehandler", so I think we can mention that middlewares are basically the same as antehandlers, with an added possibility of performining post tx-handler transaction.

Reply:
Sure, will add it!

Reply:
added